lib/alchemy/resource_filter.rb

```diff
@@ -0,0 +1,39 @@
+module Alchemy
+  class ResourceFilter
+    attr_reader :name
+
+    def initialize(filter, resource_name)
+      @filter = filter
+      @name = filter[:name]
+      @resource_name = resource_name
+      @values = filter[:values].blank? ? [] : filter[:values]
+    end
+
+    def values
+      if pretranslated?
+        @values.map { |v| v[1] }
+      else
+        @values
+      end
+    end
+
+    def translated_values
+      if pretranslated?
+        @values.map { |a| a[0] }
+      else
+        @values.map { |v| Alchemy.t(v.to_sym, scope: ["filters", @resource_name, @name, "values"]) }
+      end
+
+    end
+
+    def options_for_select
+      translated_values.zip(values)
+    end
```

Comment:
I think this is the only public method we need in this service class. All other methods are only consumed by this.